cpp/src/link_analysis/hits_impl.cuh

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <thrust/fill.h>
+#include <thrust/transform.h>
+#include <cugraph/algorithms.hpp>
+#include <cugraph/graph_view.hpp>
+#include <cugraph/prims/copy_to_adj_matrix_row_col.cuh>
+#include <cugraph/prims/copy_v_transform_reduce_in_out_nbr.cuh>
+#include <cugraph/prims/count_if_v.cuh>
+#include <cugraph/prims/reduce_v.cuh>
+#include <cugraph/prims/row_col_properties.cuh>
+#include <cugraph/prims/transform_reduce_v.cuh>
+
+// TODO : delete
+#include <string>
+
+namespace cugraph {
+namespace detail {
+template <typename GraphViewType, typename result_t>
+void normalize(raft::handle_t const& handle,
+               GraphViewType const& graph_view,
+               result_t* hubs,
+               raft::comms::op_t op)
+{
+  auto hubs_norm = reduce_v(handle,
+                            graph_view,
+                            hubs,
+                            hubs + graph_view.get_number_of_local_vertices(),
+                            identity<result_t>(op),
+                            op);
```

Comment:
Just to double check, hubs_norm can't be 0, right? If it can be 0, the transform below can be problematic and we need to be able to handle the case separately.

Reply:
Norm cannot be 0. I can put a CUGRAPH_EXPECTS to throw an exception.